src/main/scala/swiss/dasch/domain/BulkIngestService.scala

```diff
@@ -78,35 +80,58 @@ final case class BulkIngestServiceLive(
     csv: Path
   ): Task[IngestResult] =
     for {
-      _           <- ZIO.logInfo(s"Ingesting file $fileToIngest")
-      simpleAsset <- AssetRef.makeNew(project)
-      original    <- storage.createOriginalFileInAssetDir(fileToIngest, simpleAsset)
+      _        <- ZIO.logInfo(s"Ingesting file $fileToIngest")
+      assetRef <- AssetRef.makeNew(project)
+      _ <- ZIO.whenZIO(FileFilters.isNonHiddenRegularFile(fileToIngest).negate)(
+             ZIO.fail(new FileNotFoundException(s"File $fileToIngest is not a regular file"))
+           )
+      _        <- storage.getAssetDirectory(assetRef).tap(Files.createDirectories(_))
+      original <- createOriginalFileInAssetDir(fileToIngest, assetRef)
       asset <- ZIO
-                 .whenCaseZIO(FileTypes.fromPath(fileToIngest)) {
-                   case FileTypes.ImageFileType => handleImageFile(fileToIngest, original, simpleAsset)
-                   case FileTypes.VideoFileType =>
-                     ZIO.fail(new NotImplementedError("Video files are not supported yet."))
-                   case FileTypes.OtherFileType =>
-                     ZIO.fail(new NotImplementedError("Other files are not supported yet."))
+                 .fromOption(SupportedFileType.fromPath(fileToIngest))
+                 .orElseFail(new IllegalArgumentException("Unsupported file type."))
+                 .flatMap { it =>
+                   it match {
+                     case SupportedFileType.StillImage => handleImageFile(original, assetRef)
+                     case SupportedFileType.Other      => handleOtherFile(original, assetRef)
+                     case SupportedFileType.MovingImage =>
+                       ZIO.fail(new NotImplementedError("Video files are not supported yet."))
+                   }
```

Comment:
you don't like `whenCaseZIO`? 😉

Reply:
I think the most simple type the `fromPath` can return is an `Option`, having a `ZIO` as a return type was what I thought could be simplified.